sdks/go/pkg/beam/io/bigqueryio/bigquery.go

```diff
@@ -184,6 +187,33 @@ type writeFn struct {
 	Type beam.EncodedType `json:"type"`
 }
 
+// Approximate the size of an element as it would appear in a BQ insert request.
+func getInsertSize(v interface{}, schema bigquery.Schema) (int, error) {
```

Comment:
It seems a little wasteful to encode the data, throw that info away, and then allow the BQ client to encode the data again, but this is certainly precise.

I imagine the reflect approach you had would have worked fine, if things like integers getting converted to strings were accounted for better, and as well as accounting for field names, among other bits of overhead.

Reply:
Yeah, this was the best I could think of. Do you prefer going down the reflect path? It feels there are more opportunities to get things wrong in that path.

Reply:
It's a precise strategy, so I'm not against it.
The IOs in the Go SDK are not in a great state to begin with, so this is a welcome improvement.

I'd much rather have something that's a bit wasteful but works, and can be improved later, than block on trying to get it perfect.